qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,698 @@
+            free_params: The parameters that should remain unassigned.
+            group: The calibration group from which to draw the
+                parameters. If not specifies this defaults to the 'default' group.
+
+        Returns:
+            schedule: A copy of the template schedule with all parameters assigned
+                except for those specified by free_params.
+
+        Raises:
+            CalibrationError: if the name of the schedule is not known or if a parameter could
+                not be found.
+        """
+        if (name, qubits) in self._schedules:
+            schedule = self._schedules[(name, qubits)]
+        elif (name, None) in self._schedules:
+            schedule = self._schedules[(name, None)]
+        else:
+            raise CalibrationError(f"Schedule {name} is not defined for qubits {qubits}.")
+
+        # Retrieve the channel indices based on the qubits and bind them.
+        binding_dict = {}
+        for ch in schedule.channels:
+            if ch.is_parameterized():
+                binding_dict[ch.index] = self._get_channel_index(qubits, ch)
+
+        # Binding the channel indices makes it easier to deal with parameters later on
+        schedule = schedule.assign_parameters(binding_dict, inplace=False)
+
+        # The following code allows us to get the keys when the schedule has call instructions.
+        # We cannot inline the subroutines yet because we would lose the name of the subroutines.
+        parameter_keys = Calibrations.get_parameter_keys(schedule, set(), binding_dict, qubits)
+
+        # Now that we have the parameter keys we must inline all call subroutines.
+        schedule = inline_subroutines(schedule)
+
+        # Build the parameter binding dictionary.
+        free_params = free_params if free_params else []
+
+        for key in parameter_keys:
+            if key.parameter not in free_params:
+                # Get the parameter object. Since we are dealing with a schedule the name of
+                # the schedule is always defined. However, the parameter may be a default
+                # parameter for all qubits, i.e. qubits may be None.
+                if key in self._parameter_map:
+                    param = self._parameter_map[key]
+                elif (key.schedule, key.parameter, None) in self._parameter_map:
+                    param = self._parameter_map[(key.schedule, key.parameter, None)]
+                else:
+                    raise CalibrationError(
+                        f"Ill configured calibrations {key} is not present and has not default value."
+                    )
+
+                if param not in binding_dict:
+                    binding_dict[param] = self.get_parameter_value(
+                        key.parameter, key.qubits, key.schedule, group=group
+                    )
+
+        return schedule.assign_parameters(binding_dict, inplace=False)
+
+    @staticmethod
+    def get_parameter_keys(
```

Comment:
I think this can be a private method because no public method requires parameter key.

Reply:
I agree. See here: 54db71c